ecs/src/nox/memory/FixedSlotPool.h

```diff
@@ -0,0 +1,134 @@
+             */
+            FixedSlotPool& operator=(const FixedSlotPool&) = delete;
+
+            /**
+             * @brief      Move constructor. Moves all members out of source.
+             *
+             * @warning    source will not be usable after moving. 
+             * 
+             * @param[in]  source  the FixedSlotPool to move from.
+             */
+            FixedSlotPool(FixedSlotPool&& source);
+
+            /**
+             * @brief      Move assignment operator. Moves all members out of
+             *             source. The class is tolerant of self-assignment.
+             *
+             * @warning    source will not be usable after moving.
+             *
+             * @param[in]  source  the FixedSlotPool to move from.
+             *
+             * @return     *this after assignment.
+             */
+            FixedSlotPool& operator=(FixedSlotPool&& source);
+
+            /**
+             * @brief      Destroys all blocks in the pool. The user must ensure
+             *             that none of the slots holds any non destructed
+             *             objects.
+             */
+            ~FixedSlotPool();
+
+            /**
+             * @brief      Allocates a slotSize sized memory area. The function
+             *             ignores the parameter, the only reason for this
+             *             parameter existing is to ease performance testing in
+             *             real world case. See:
+             *             https://github.com/Per-Morten/imt3912_dev/issues/81
+             *
+             * @param[in]  size  Ignored!
+             *
+             * @return     Pointer to allocated uninitialized memory.
+             */
+            void* allocate(std::size_t size);
+
+            /**
+             * @brief      Function is ignored in this allocator. The function
+             *             exists only reason for easing performance testing in
+             *             real world case. See:
+             *             https://github.com/Per-Morten/imt3912_dev/issues/81
+             *
+             * @param      ptr   Ignored!
+             */
+            void deallocate(void* ptr) {}
+
+            /**
+             * @brief      Resets all the slots to 0. User must ensure that
+             *             delete has been called on all stored elements first.
+             */ 
+            void reset();
+
+        private:
+            /**
+             * @brief      Block holding a link to the next block, as well as
+             *             the memory area within the block.
+             */ 
+            struct Block
+            {
+                /**
+                 * @brief      Pointer to next block within the list of blocks.
+                 */
+                Block* next{};
+
+                /**
+                 * @brief      How many bytes that is used within the slots.
+                 */
+                std::size_t used{0};
```

Comment:
Not agreeing as I am initializing arguments, and initializing with {} is not different than initializing with any other arguments. 
Example:
new int(3);
new int{ 3 }; // Seems weird for me to change that.